src_assets/windows/assets/shaders/directx/SceneVS.hlsl

```diff
@@ -1,22 +1,37 @@
 struct PS_INPUT
 {
-	float4 pos : SV_POSITION;
-	float2 tex : TEXCOORD;
+    float4 pos : SV_POSITION;
+    float2 tex : TEXCOORD;
+};
+
+cbuffer rotation_info : register(b1) {
+    int rotation;
 };
 
 PS_INPUT main_vs(uint vI : SV_VERTEXID)
 {
-	float idHigh = float(vI >> 1);
-	float idLow = float(vI & uint(1));
+    PS_INPUT output;
 
-	float x = idHigh * 4.0 - 1.0;
-	float y = idLow * 4.0 - 1.0;
+    if (vI == 0) {
+        output.pos = float4(-1, -1, 0, 1);
+        output.tex = float2(0, 1);
+    }
+    else if (vI == 1) {
+        output.pos = float4(-1, 3, 0, 1);
+        output.tex = float2(0, -1);
+    }
+    else if (vI == 2) {
+        output.pos = float4(3, -1, 0, 1);
+        output.tex = float2(2, 1);
+    }
 
-	float u = idHigh * 2.0;
-	float v = 1.0 - idLow * 2.0;
+    if (rotation != 0) {
+        float rotation_radians = radians(90 * rotation);
+        float2x2 rotation_matrix = { cos(rotation_radians), -sin(rotation_radians),
+                                     sin(rotation_radians), cos(rotation_radians) };
+        float2 rotation_center = { 0.5, 0.5 };
+        output.tex = round(rotation_center + mul(rotation_matrix, output.tex - rotation_center));
```

Comment:
Can we precompute these values and put them in the constant buffer?

Reply:
We can, but this is in the vertex shader that runs for 3 vertices. So maybe better not prematurely optimize it. Same reason why I replaced bit-shift magic with easier to understand `if` branches.

Reply:
Normally I'd agree, but in this case we're already creating the constant buffer to store the rotation angle and the matrix logic ends up getting duplicated in several shaders. It seems like moving it to C++ would be cleaner and faster.

Reply:
That's two separate problems, we can rule "faster", 5 basic math functions called for 3 vertices two or three times per frame is literally nothing for gpu.

Now for the duplication, this one is legitimate concern. But if we move the rotation to generic matrix defined externally to the shader, it will start implying that we support arbitrary angle rotation and rounding part will stop making sense.
`output.tex = round(rotation_center + mul(rotation_matrix, output.tex - rotation_center));`
So maybe we can do de-duplication in shader? By using some `common.hlsl` and including it around.

Reply:
Ok, I don't feel very strongly about this. If you'd much rather leave it as is or change it to use an HLSL include to deduplicate the code, either is fine with me.

Reply:
Can this be done in separate pull request?
There are more things that can be deduplicated other than this rotation matrix, might as well tackle all of them.

Reply:
Yeah that's fine with me.